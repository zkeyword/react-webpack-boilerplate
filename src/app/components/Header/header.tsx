import { stringify } from 'query-string'
import * as React from 'react'
import { RouteComponentProps, withRouter } from 'react-router-dom'
import './header.styl'

interface IProps extends RouteComponentProps {
    type?: string[]
    systemName?: string
    children: JSX.Element[] | JSX.Element
}

const Header: React.FunctionComponent<IProps> = (props: IProps): React.ReactElement => {
    const jump = (target: string): void => {
        props.history.push(target)
    }
    return (
        <header className="lt-header">
            <div className="logo" onClick={() => jump('/')}></div>
            {props.children}
        </header>
    )
}

export default withRouter(Header)
