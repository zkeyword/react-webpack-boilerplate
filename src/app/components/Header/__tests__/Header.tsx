import * as React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import Header from '../index'

describe('<Header />', () => {
    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <Header>
                    <div className="text" />
                </Header>
            </MemoryRouter>
        ).container.firstChild
        expect(tree).toMatchSnapshot()
    })
    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        ).container.firstChild
        expect(tree).toMatchSnapshot()
    })
})
