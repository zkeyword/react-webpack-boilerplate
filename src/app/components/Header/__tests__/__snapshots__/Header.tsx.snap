// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<header
  class="header"
>
  <div
    class="logo"
  />
  <div
    class="text"
  />
</header>
`;

exports[`<Header /> renders 2`] = `
<header
  class="header"
>
  <div
    class="logo"
  />
</header>
`;
